specification/azureactivedirectory/resource-manager/Microsoft.Aadiam/stable/2017-04-01/azureactivedirectory.json

```diff
@@ -0,0 +1,557 @@
+        }
+      },
+      "required": [ "enabled", "days" ],
+      "description": "Specifies the retention policy for the log."
+    },
+    "MetricSettings": {
+      "properties": {
+        "timeGrain": {
+          "type": "string",
+          "format": "duration",
+          "description": "The timegrain of the metric in ISO8601 format."
+        },
+        "category": {
+          "type": "string",
+          "description": "Name of a Diagnostic Metric category for a resource type this setting is applied to. To obtain the list of Diagnostic metric categories for a resource, first perform a GET diagnostic settings operation."
+        },
+        "enabled": {
+          "type": "boolean",
+          "description": "A value indicating whether this category is enabled."
+        },
+        "retentionPolicy": {
+          "$ref": "#/definitions/RetentionPolicy",
+          "description": "The retention policy for this category."
+        }
+      },
+      "required": [ "enabled" ],
+      "description": "Part of MultiTenantDiagnosticSettings. Specifies the settings for a particular metric."
+    },
+    "LogSettings": {
+      "properties": {
+        "category": {
+          "type": "string",
+          "description": "Name of a Diagnostic Log category for a resource type this setting is applied to. To obtain the list of Diagnostic Log categories for a resource, first perform a GET diagnostic settings operation."
+        },
+        "enabled": {
+          "type": "boolean",
+          "description": "A value indicating whether this log is enabled."
+        },
+        "retentionPolicy": {
+          "$ref": "#/definitions/RetentionPolicy",
+          "description": "The retention policy for this log."
+        }
+      },
+      "required": [ "enabled" ],
+      "description": "Part of MultiTenantDiagnosticSettings. Specifies the settings for a particular log."
+    },
+    "DiagnosticSettings": {
+      "properties": {
+        "storageAccountId": {
+          "type": "string",
+          "description": "The resource ID of the storage account to which you would like to send Diagnostic Logs."
+        },
+        "serviceBusRuleId": {
+          "type": "string",
+          "description": "The service bus rule Id of the diagnostic setting. This is here to maintain backwards compatibility."
+        },
+        "workspaceId": {
+          "type": "string",
+          "description": "The workspace ID (resource ID of a Log Analytics workspace) for a Log Analytics workspace to which you would like to send Diagnostic Logs. Example: /subscriptions/4b9e8510-67ab-4e9a-95a9-e2f1e570ea9c/resourceGroups/insights-integration/providers/Microsoft.OperationalInsights/workspaces/viruela2"
+        },
+        "eventHubAuthorizationRuleId": {
+          "type": "string",
+          "description": "The resource Id for the event hub authorization rule."
+        },
+        "eventHubName": {
```

Comment:
why is this not the fully qualified resource id? Specially when user can provide an existing one.

Reply:
Field seems to be there just so the user can provide a different name for what's referenced in eventHubAuthorizationRuleId if they want to, but doesn't seem to get used in any other way but as a label. I'll look into removing it on the next API iteration.